### PR TITLE
More dreamhost compatibility

### PR DESCRIPTION
**Pull request by jbtule**
- added some more packages that are in dreamhost apt-get list on a shared plan (git & ruby.18) though not exhaustive
 - added a public folder on the variable config, many project types need a public subfolder to be hosted by apache rather than the full project root
 - synced folder can't chown root since NFS, modified vagrant file to add www-data
 - provision user based on dreambox_user_name, vagrant ssh then uses that user.

### CODE REVIEW

**Comment by PabloScolpino** [comment]
+1 for this PR

**Comment by goodguyry** [comment]
I agree!

I've been sidelined for a bit, but hoping to get back on this ASAP.

Thanks for the work on this, @jbtule! Sorry I left you hanging for so long.

File: files/user_setup [hunk 1]
```diff
@@ -160,7 +171,7 @@ if [[ $status -eq 2 ]]; then
   echo
   echo "Enter the name of the directory within which DreamBox will find your"
   echo "site's files, relative to your Vagrantfile (or press return for '$DEFAULT_PROJECT_DIR')."
-  echo
+esca  echo
```
Comment (goodguyry) [inline files/user_setup]: This is either a typo or some fancy Bash trick I'm not familiar with.

File: templates/Vagrantfile [hunk 2]
```diff
@@ -1,13 +1,25 @@
...
+  
+  $user_vars["DREAMBOX_UID"] = 31415
 
   # Run user_setup
   config.vm.provision 'shell',
     inline: '/bin/bash /usr/local/bin/user_setup',
     # Pass user_setup ENV variables to this script
     :env => $user_vars
+
+  if VAGRANT_COMMAND == "ssh"
```
Comment (goodguyry) [inline templates/Vagrantfile:15]: Interesting! I was moving toward dropping the user creation stuff, but this may have just changed my mind.

**Review by goodguyry — CHANGES_REQUESTED** [review]
I'm sure you've tested this and it works fine for you, but I'm just wondering about the `esca` line. Thanks!

File: templates/Vagrantfile [hunk 3]
```diff
@@ -1,13 +1,25 @@
...
+      config.ssh.username = $user_vars['DREAMBOX_USER_NAME']
```
Comment (goodguyry) [inline templates/Vagrantfile]: The indentation is off a bit here.

**Review by goodguyry — APPROVED** [review]

**Pull request merged**